modules/core.nix

```diff
@@ -248,5 +276,27 @@ let cfg = config.nix-dabei; in
         initrd-parse-etc.enable = false;
       };
     })
+
+
+    # Synchronize time using NTP to prevent clock skew that could interfere
+    # with date & time sensitive operations like certificate verification.
+    (lib.mkIf cfg.ntpSync.enable {
+      boot.initrd.systemd.services.ntpdate-timesync = let
+       ntpServersAsString = lib.concatStringsSep " " cfg.ntpSync.servers;
+     in {
+        requires = [ "initrd-fs.target" "network-online.target"];
+        requiredBy = [ "auto-installer.service" ];
+        before = [ "auto-installer.service" ];
+        after = [ "initrd-fs.target" "network-online.target"];
+        unitConfig.DefaultDependencies = false;
+        serviceConfig.Type = "oneshot";
+        script = ''
+            ntpdate -b ${ntpServersAsString}
+            if [[ "${lib.boolToString cfg.ntpSync.updateHwClock}" == "true" ]]; then
```

Comment:
I believe you could replace that if statement with `${lib.optionalString cfg.ntpSync.updateHwClock "hwclock --systohc"}`. 

Otherwise LGTM!

Reply:
Thx, agreed this makes a much better syntax. Done in [8e6b46a](https://github.com/dep-sys/nix-dabei/pull/19/commits/8e6b46a4af5c9ba372477da5257bde05e19afffa)